common/src/main/scala/org/neo4j/spark/service/SchemaService.scala

```diff
@@ -149,7 +148,7 @@ class SchemaService(
     params: java.util.Map[String, AnyRef],
     extractFunction: Record => Map[String, AnyRef]
   ): mutable.Buffer[StructField] = {
-    session.run(query, params, sessionTransactionConfig).list.asScala
+    session.executeRead(tx => tx.run(query, params).list.asScala, sessionTransactionConfig)
```

Comment:
Its a very small thing, but just wanted to note that `asScala` is called outside the function in other similar places.